src/AutoMapper/TypeMap.cs

```diff
@@ -76,14 +76,7 @@ public static Exception MissingMapException(Type sourceType, Type destinationTyp
     public MemberList ConfiguredMemberList
     {
         get => (_details?.ConfiguredMemberList).GetValueOrDefault();
-        set
-        {
-            if (value == default)
```

Comment:
The check is here to avoid creating `_details` in the default case for `CreateMap`. So we want that _and_ your use case to work. Somehow :)